File: ZpCsTest/ZPTest.cs
﻿using System;
using System.Drawing;
using System.Globalization;
using System.Threading;
using System.Threading.Tasks;
using ZennoLab.CommandCenter;
using ZennoLab.InterfacesLibrary.ProjectModel;

namespace ZPTest
{
    public class Main : IZennoCustomCode, IZennoCustomEndCode
    {
        public int ExecuteCode(Instance instance, IZennoPosterProjectModel project)
        {
            project.SendInfoToLog("Test");

            return 0;
        }

        public void GoodEnd(Instance instance, IZennoPosterProjectModel project) { }

        public void BadEnd(Instance instance, IZennoPosterProjectModel project) { }
    }
}

File: ZpFsTest/ZpTest.fs
namespace ZpTest

open System
open ZennoLab.CommandCenter
open ZennoLab.InterfacesLibrary.ProjectModel

type public Main() =
    interface IZennoCustomCode with
        member this.ExecuteCode(instance: Instance, project: IZennoPosterProjectModel): int =
            project.SendInfoToLog "Test"
            0

    interface IZennoCustomEndCode with
        member this.GoodEnd(instance: Instance, project: IZennoPosterProjectModel): unit = ()
        member this.BadEnd(instance: Instance, project: IZennoPosterProjectModel): unit = ()
